x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_list/components/alerts_list.tsx

```diff
@@ -220,40 +222,52 @@ export const AlertsList: React.FunctionComponent = () => {
   }, []);
 
   async function loadAlertsData() {
-    const hasAnyAuthorizedAlertType = alertTypesState.data.size > 0;
-    if (hasAnyAuthorizedAlertType) {
-      setAlertsState({ ...alertsState, isLoading: true });
-      try {
-        const alertsResponse = await loadAlerts({
-          http,
-          page,
-          searchText,
-          typesFilter,
-          actionTypesFilter,
-          alertStatusesFilter,
-          sort,
-        });
-        await loadAlertAggs();
-        setAlertsState({
-          isLoading: false,
-          data: alertsResponse.data,
-          totalItemCount: alertsResponse.total,
-        });
+    if (alertTypesState.isInitialized) {
```

Comment:
Does this open the door for any weird behavior with race conditions? Is it possible to call this function _before_ this variable is true and what happens in that scenario? Will the UI filters not be properly represented in the data? 

Looking through the code, it doesn't seem possible for this to be false?

Reply:
`loadAlertsData` is called in a `useEffect` that has `alertTypesState` as a dependency

```
useEffect(() => {
    loadAlertsData();
  }, [
    alertTypesState,
    page,
    searchText,
    JSON.stringify(typesFilter),
    JSON.stringify(actionTypesFilter),
    JSON.stringify(alertStatusesFilter),
  ]);
```

so if it happens to be called before `alertTypesState. isInitialized` is true, then it should retrigger the fn call when `alertTypesState. isInitialized` is set to true.

The only way `alertTypesState. isInitialized` would never be set to true is if the API call threw an error. In that case, it looks like the behavior (which this PR doesn't change) is to show a toast and a forever loading spinner 😬 I'll create a new issue for that

Reply:
Thanks for clarifying!